src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+
+    private static final String EXAMPLE_DATA = "C56A4180-65AA-42EC-A945-5FD21DEC0538";
+
+    public ActivityServiceImpl(OrganizationService organizationService, ConflictService conflictService) {
+        this.organizationService = organizationService;
+        this.conflictService = conflictService;
+    }
+
+    @Override
+    public List<ActivityDTO> getAllOrganizationActivities(UUID systemAccountId) {
+        List<ActivityDTO> activities = new ArrayList<>();
+        List<OrganizationDTO> orgs = organizationService.findAllWithOwnerId(systemAccountId);
+
+        // TODO: get organization-entities mapping for every organization
+        // to-remove-start
+        // mock for int tests will be removed when Organization matching will be available
+        Map<UUID, UUID> orgToEntity = new HashMap<>();
+        orgToEntity.put(orgs.get(0).getId(), UUID.fromString(EXAMPLE_DATA));
+        // to-remove-end
+
+        for (Map.Entry<UUID, UUID> entry : orgToEntity.entrySet()) {
+            Optional<ActivityDTO> activityOpt = getEntityActivity(entry.getKey(), entry.getValue());
+            activityOpt.ifPresent(activity -> activities.add(activity));
+        }
+
+        return activities;
+    }
+
+    @Override
+    public Page<ActivityDTO> getAllOrganizationActivities(Pageable pageable, UUID systemAccountId) {
+        List<ActivityDTO> activities = getAllOrganizationActivities(systemAccountId);
+
+        int pageSize = pageable.getPageSize();
+        int currentPage = pageable.getPageNumber();
+        int startItem = currentPage * pageSize;
+        List<ActivityDTO> list;
+
+        if (activities.size() < startItem) {
+            list = Collections.emptyList();
+        } else {
+            int toIndex = Math.min(startItem + pageSize, activities.size());
+            list = activities.subList(startItem, toIndex);
+        }
+
+        return new PageImpl<>(list, PageRequest.of(currentPage, pageSize), activities.size());
+    }
+
+    private Optional<ActivityDTO> getEntityActivity(UUID orgId, UUID resourceId) {
+        log.debug("Creating Activity for organization: {} with resourceId: {}", orgId, resourceId);
+        try {
+            Optional<OrganizationDTO> opt = organizationService.findOne(orgId);
+            OrganizationDTO org = opt.orElseThrow(() -> new ActivityCreationException(
+                String.format("There are no conflicts for resourceId: %s", orgId)));
+
+            List<ConflictDTO> conflictDTOS = conflictService.findAllWithResourceIdAndOwnerId(resourceId, org.getAccountId());
+            if (conflictDTOS != null && conflictDTOS.isEmpty()) {
+                throw new ActivityCreationException(String.format("There are no conflicts for resourceId: %s", resourceId));
+            }
+
+            return Optional.of(ActivityDTO.builder()
+                .conflicts(conflictDTOS)
+                .organization(org)
+                .build());
+        } catch (ActivityCreationException ex) {
```

Comment:
Please do not catch the exception here, but throw it, or return empty() instead

Reply:
ok